solution.cpp

```diff
@@ -0,0 +1,168 @@
+/* ----------------------------------------------------------------------------
+ * Copyright &copy; 2015 Jaemin Jin <jaemin_95@fullerton.edu>
+ * Released under the [MIT License] (http://opensource.org/licenses/MIT)
+ * ------------------------------------------------------------------------- */
+
+/**
+ * Do recursion/iteration for the following: Euclid's algorithm, Fibonacci, Roman Numerals, and Integers to Words.
+ */
+
+#include <iostream>
+using std::cout;
+using std::cin;
+using std::endl;
+
+#include <string>
+using std::string;
+
+// #1: Function Prototypes for "Euclid's Algorithm (GCD)"
+
+// int gcd_iterative(int a, int b);
+// int gcd(int a, int b);
+// int abs(int a);
+
+// #1: Function Definitions
+
+// I did this function on my own this time.
+int gcd_iterative(int a, int b)                                 // Iterative form
+{
+    a = abs(a);
+    b = abs(b);
+    
+    while (a == 0 && b == 0)
+        if (a > b)
+            a -= b;
+        else
+            b -= a;
+    return a+b;
+}
+
+// Referenced during the lecture(meeting-11-1.mp4).
+int gcd(int a, int b)                                           // Recursive form
+{
+    a = abs(a);
+    b = abs(b);
+    
+    if (a == 0 || b == 0)
+        return a+b;
+    
+    if (a > b)
+        return gcd(a-b, b);
+    else
+        return gcd(a,b-a);
+}
+
+int abs(int a)
+{
+    return (a >= 0) ? a: -a;
+}
+
+// #2: Function Prototypes for Fibonacci
+
+// int fibonacci_iterative(int n, int f1 = 1, int f2 = 1);
+// int fibonacci(int n, int f1 = 1, int f2 = 1);
+
+// #2: Function Definitions
+
+// Referenced and rephrased "Fibonacci in C++ " on http://talkbinary.com/programming/c/fibonacci-in-c/ after many serious attempts on my own these functions.
+int fibonacci_iterative(int n, int f1 = 1, int f2 = 1)           // Iterative form
+{
+    if ( n == 1 || n == 2)
+        return f1 || f2;
```

Comment:
this will return `1` if either of `f1` or `f2` is nonzero, and `0` if both are zero, which is probably not quite what you want (what if one of `f1` and `f2` is greater than 1?)